dll/cpl/access/general.c

```diff
@@ -194,8 +195,10 @@ GeneralPageProc(HWND hwndDlg,
                            IDC_RESET_BOX,
                            pGlobalData->accessTimeout.dwFlags & ATF_TIMEOUTON ? BST_CHECKED : BST_UNCHECKED);
             FillResetComboBox(GetDlgItem(hwndDlg, IDC_RESET_COMBO));
-            SendDlgItemMessage(hwndDlg, IDC_RESET_COMBO, CB_SETCURSEL,
-                               (pGlobalData->accessTimeout.iTimeOutMSec / 300000) - 1, 0);
+            iCurSel = (pGlobalData->accessTimeout.iTimeOutMSec / 300000) - 1;
+            if (iCurSel < 0)
+                iCurSel = 0;
```

Comment:
Maybe somewhat more explicit:
```suggestion
            iCurSel = pGlobalData->accessTimeout.iTimeOutMSec != 0
                      ? (pGlobalData->accessTimeout.iTimeOutMSec / 300000) - 1 : 0;
```
Or would it be possible to do something like 'pGlobalData->accessTimeout.iTimeOutMSec = 300000' somewhere (in `ReadSettings()`?) instead?

Reply:
Your suggested code is buggy (if timeout is less than 300 seconds but non-zero, the bug would still happen).
But it could be adapted as follows:
```suggestion
            pGlobalData->accessTimeout.iTimeOutMSec = max(pGlobalData->accessTimeout.iTimeOutMSec, 300000);
            iCurSel = (pGlobalData->accessTimeout.iTimeOutMSec / 300000) - 1;
```
The first line implements the "something like ...." suggestion in a correct way.

Reply:
> Your suggested code is buggy (if timeout is less than 300 seconds but non-zero, the bug would still happen).

Is it? Is that example case possible?
If it is, then that would strengthen my suggestion to validate this value(s) earlier than in "UI" code.

Reply:
That iTimeOutMSec from the registry can be set to less than 300000 is perfectly fine (the 5-minute increase chunks is just a limitation of this GUI screen, but that's just it.)

Reply:
Imho the current
![image](https://github.com/reactos/reactos/assets/33393466/b40425a0-fa4b-4aa6-bed4-fcffc08f4f2d)
is easier to read (and also more explicit) than your suggestion Serge.
So in case it is not broken, I would vote to keep it like it is.

Reply:
> iTimeOutMSec from the registry

I assume that value should still not be changed here, should it?

> limitation of this GUI screen

Ah, ok then.

Reply:
> easier to read (and also more explicit)

Imho: easier yes, explicit "no".

> not broken

Both give the same result.